### PR TITLE
Invoice View

### PR DESCRIPTION
**Pull request by galitskyd**
On the overview: replace "Send invoice" with "Details"
On Purchased Products: replace "View Jobs" with "Details"
Od Purchased Product Details: Add "View Invoice" link to footer
On View Invoice Page add "Send me this invoice" button in right nav.
Add print functionality.

### CODE REVIEW

File: postajob/models.py [hunk 1]
```diff
@@ -938,7 +938,7 @@ class Invoice(BaseModel):
     # Owner is the Company that owns the Products.
     owner = models.ForeignKey('seo.Company', related_name='owner')
 
-    def send_invoice_email(self, other_recipients=None):
+    def send_invoice_email(self, send_to_admins=True, other_recipients=None):
         """
         Sends the invoice email to the company admins along with
         any other optional recipients.
```
Comment (almathew) [inline postajob/models.py]: This comment is no longer true for all uses of the function. We should probably update to actually explain kwargs.

File: postajob/views.py [hunk 2]
```diff
@@ -63,6 +63,28 @@ def view_job(request, purchased_product, pk, admin):
     return render_to_response('postajob/%s/view_job.html' % settings.PROJECT,
                               data, RequestContext(request))
 
+@company_has_access(None)
+def view_invoice(request, purchased_product):
+    company = get_company_or_404(request)
+    product = PurchasedProduct.objects.get(pk=purchased_product)
```
Comment (almathew) [inline postajob/views.py]: You might want to add `owner=company` here and use `get_object_or_404()`

**Comment by almathew** [comment]
@galitskyd I know it's not your fault, but can you use this branch to fix my typo (`invoice.transcation`) on line 66 of invoice_email.html?

**Comment by aspidites** [comment]
Front end looks fine. Everything operates as described. Just need to review actual code, unless someone else already has.

File: postajob/views.py [hunk 3]
```diff
@@ -63,6 +63,27 @@ def view_job(request, purchased_product, pk, admin):
...
+    product = get_object_or_404(PurchasedProduct, pk=purchased_product,
+                                owner=company)
+    invoice = product.invoice
+    data = {
+        'company': company,
+        'purchased_product': product,
+        'invoice': invoice,
+        'purchases': invoice.invoiced_products.all()
+    }
+    if request.GET.get('m'):
```
Comment (aspidites) [inline postajob/views.py]: Can we document m or give it a more meaningful name?

File: templates/postajob/dseo/view_invoice.html [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    <label>Job Limit:</label>
+    <b>{% if purchased_product.num_jobs_allowed == 0 %}Unlimited{%else%}{{ purchased_product.num_jobs_allowed }} ({{purchased_product.jobs_remaining}} left){%endif%}</b>
+    <br /><label>Purchase Date:</label>
+    <b>{{ purchased_product.purchase_date }}</b>
+    <br/><label>Expires:</label>
+    <b>{{ purchased_product.expiration_date }}</b>
+    <br />
+    <div class="navigation">
+        <h3>Navigation</h3>
+        <a class="btn" href="{% url 'resend_invoice' pk=purchased_product.pk %}">Send me this Invoice</a>
```
Comment (aspidites) [inline templates/postajob/dseo/view_invoice.html]: Send Me This Invoice -> http://titlecase.com/

**Pull request merged**